src/history.ts

```diff
@@ -114,11 +114,12 @@ export default class History {
   constructor(protected options: HistoryInstallOptions) {
   }
 
-  push(event: Event) {
+  push(event: Event, onPushed?: Function) {
     if (this.options.filter && !this.options.filter(event)) {
       return
     }
     this.items.push(event)
+    onPushed && onPushed()
   }
```

Comment:
Habe aktuell zu wenig Überblick, wie pushed genutzt wird. Wäre hier ein async/await Ansatz sinnvoll?
`super.push(event).then(() => {
if (this.options.onEvent) {
      this.options.onEvent(event)
    }
    if (this.options.feed) {
      this.logNewEvent(event)
    }	
})`

Reply:
@ma-jahn there is no asynchronous flow. A synchronous alternative to`onPushed` would be to return a boolean (whether or not the item was filtered), but that would increase the complexity of all sub classes, because in TS they would be forced to satisfy the interface of the parent class (boolean return value). I tried that variant and it looked much more complex to me.